### ISSUE TITLE
bin/admin/build-news builds news for web2 even though web2 no longer has releases

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4190]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#4190|
|Type|defect|
|Created|2009-12-25 19:12:36Z|
|Branch|https://github.com/twisted/twisted/tree/news-blacklist-4190|

It shouldn't generate web2 news.

<details><summary>Searchable metadata</summary>

```
trac-id__4190 4190
type__defect defect
reporter__radix radix
priority__normal normal
milestone__ 
branch__branches_news_blacklist_4190 branches/news-blacklist-4190
branch_author__jml jml
status__closed closed
resolution__fixed fixed
component__release_management release management
keywords__ 
time__1261768356000000 1261768356000000
changetime__1275846918000000 1275846918000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_1"></a>|@jml commented|
|-|-|

(In [29219]) Branching to 'news-blacklist-4190'

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_2"></a>|@jml commented|
|-|-|

(In [29222]) Do not generate news for blacklisted projects. refs [#4190](https://github.com/twisted/twisted/issues/4190).

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_3"></a>|@jml removed owner|
|-|-|

In this branch, I moved out the literal value of `DistributionBuilder.blacklist` to a new top-level variable `PROJECT_BLACKLIST` and made `DistributionBuilder.blacklist` equal to that.

`NewsBuilder` gets a similar class variable, `blacklist`. It's only consulted in `buildAll`.

For testing, I just added a blacklisted project to the existing test for `buildAll`, since I couldn't see any value in adding a new test and since I couldn't think of a more unit-y test.

If this branch lands, the current branch for [#4138](https://github.com/twisted/twisted/issues/4138) will need to be updated. I had thought of adding a `blacklisted` property to project, but that would make [#4138](https://github.com/twisted/twisted/issues/4138) even more difficult to land.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @jml|
|-|-|

Looks great, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_5"></a>|@jml set status to `closed`|
|-|-|

(In [29224]) Merge news-blacklist-4190

 * Author: jml
 * Reviewer: therve
 * Fixes [#4190](https://github.com/twisted/twisted/issues/4190)

Do not generate news sections for blacklisted projects.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
